kube_hunter/conf/__init__.py

```diff
@@ -7,12 +7,15 @@
 
 loglevel = getattr(logging, config.log.upper(), None)
 
-if not loglevel:
-    logging.basicConfig(level=logging.INFO,
-                        format=formatter)
+# If the user submitted a log option that isn't none or a log level, default to
+# logging.INFO
+if config.log.lower() != "none" and not loglevel:
     logging.warning('Unknown log level selected, using info')
-elif config.log.lower() != "none":
-    logging.basicConfig(level=loglevel,
-                        format=formatter)
+
+    loglevel = logging.INFO
+
+# If the user didn't submit a log level of none, set the log level
+if loglevel is not None:
+    logging.basicConfig(level=loglevel, format=formatter)
```

Comment:
It will be cleaner to disable logging this way:
```suggestion
log_level_name = config.log.upper()

if log_level_name == "NONE":
    logging.disable(logging.CRITICAL)
else:
    log_level = getattr(logging, log_level_name, None)
    if not log_level:
        log_level = logging.INFO
        logging.warning("Unknown log level selected, using INFO")
    logging.basicConfig(level=loglevel, format=formatter)
```

You can make it testable by creating a `setup_logger` function with log level name as parameter.
By doing this you can check cases of valid log level, invalid log level, and none log level (i.e., check that `root.manager.disable` is set to logging.CRITICAL).